eureka2-ext/eureka2-eureka1x-rest-api/src/main/java/com/netflix/eureka2/eureka1x/rest/query/AbstractEureka2RegistryView.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.eureka2.eureka1x.rest.query;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotifications;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.functions.Func1;
+import rx.subjects.BehaviorSubject;
+
+/**
+ * @author Tomasz Bak
+ */
+public abstract class AbstractEureka2RegistryView<T> {
+
+    private final Observable<ChangeNotification<InstanceInfo>> notifications;
+    private final long refreshIntervalMs;
+    private final Scheduler scheduler;
+
+    private final BehaviorSubject<T> latestCopySubject = BehaviorSubject.create();
+    private Subscription subscription;
+
+    protected AbstractEureka2RegistryView(Observable<ChangeNotification<InstanceInfo>> notifications,
+                                          long refreshIntervalMs,
+                                          Scheduler scheduler) {
+        this.notifications = notifications;
+        this.refreshIntervalMs = refreshIntervalMs;
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Connect happens immediately after child object is constructed.
+     */
+    protected void connect() {
+        subscription = notifications
```

Comment:
Does this mean that all the sub classes all have their individual subscriptions to the eureka2 registry? Would that introduce synchronisation differences between the different views? What about just having a single converter that converts from eureka2 registry interest to a eureka1 registry (store in app groups) at a regular interval, then feel all queries from there?